base_product_mass_addition/models/product_product.py

```diff
@@ -0,0 +1,107 @@
+# © 2014 Today Akretion
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# @author Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models
+from odoo.tests.common import Form
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    qty_to_process = fields.Float(
+        compute='_compute_process_qty',
+        inverse='_inverse_set_process_qty',
+        help="Set this quantity to create a new line "
+             "for this product or update the existing one."
+    )
+
+    def _prepare_quick_line(self, parent):
+        res = self._get_quick_line_qty_vals()
+        res.update({'product_id': self.id})
+        return res
+
+    def _get_quick_line(self, parent):
+        return None
+
+    def _add_quick_line(self, parent, line_model):
+        vals = self._prepare_quick_line(parent)
+        vals = self._complete_quick_line_vals(parent, vals)
+        if not vals.get('price_unit'):
+            vals['price_unit'] = 0.0
+        self.env[line_model].create(vals)
+
+    def _update_quick_line(self, parent, line):
+        if self.qty_to_process:
+            # apply the on change to update price unit if depends on qty
+            vals = self._get_quick_line_qty_vals()
+            vals['product_id'] = line.product_id.id
+            vals = self._complete_quick_line_vals(parent, vals)
+            line.write(vals)
+        else:
+            line.unlink()
+
+    def _get_quick_line_qty_vals(self):
+        return None
+
+    def _complete_quick_line_vals(self, parent, vals,
+                                  parent_key='', lines_key=''):
```

Comment:
Not blocking for me but seems the common way to define default with no value is `None`

https://stackoverflow.com/questions/13075044/initialize-parameter-of-method-with-default-value

and never `[]` FYI
https://effbot.org/zone/default-values.htm